### PR TITLE
MOSIP-44837 - Automated the testcases of bio-extractor requests endpoint

### PR DESCRIPTION
**Pull request by mohanachandran-s**
MOSIP-44837 - Automated the testcases of bio-extractor requests endpoint

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive guidance for repository setup, execution workflows, testing architecture, and AI tool usage.
  * Added usage notes for running tests, configuration options, and role/catalog references.

* **Tests**
  * Significantly expanded credential-mapping test coverage with many edge cases, error conditions, and role-based scenarios.
  * Added a bio-extractor smoke test for a valid policy case.

* **Templates**
  * Minor test result template updated to include a top-level response field.

### CODE REVIEW

File: api-test/src/main/resources/pms/MapPartnerAndPolicytoCredentialType/MapPartnerAndPolicytoCredentialType.yml [hunk 1]
```diff
@@ -170,4 +170,382 @@ MapPartnerAndPolicytoCredentialType:
...
+    "partnerPolicyRequestId": "$ID:partnerPolicyMapping_ForDeactivateOIDCNeg_All_Valid_Smoke_sid_mappingkey$",
+    "requesttime": "$TIMESTAMP$"
+}'
+      output: '{
+   "errors": [
+    {
+      "errorCode": "PMS_PRT_005"
+    }
+  ]
+}'
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply api-test/src/main/resources/pms/MapPartnerAndPolicytoCredentialType/MapPartnerAndPolicytoCredentialType.yml:285]: Ignore

**Review by jayesh12234 — APPROVED** [review]

**Review by lsivanand — APPROVED** [review]

**Review by Prafulrakhade — APPROVED** [review]
